### PR TITLE
mission display infos

### PR DESCRIPTION
**Pull request by TheEmidee**
- **Added Name and description to the mission data**
- **Added delegates on the mission system to let BPs know when mission and objective start and end**
- **Added CompleteObjective function**
- **Renamed delegates / events**
- **Created view model for UMG**

### CODE REVIEW

File: Source/MissionSystem/Private/MSMissionSystem.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include "MSMission.h"
 #include "Serialization/MemoryWriter.h"
 #include "Serialization/ObjectAndNameAsStringProxyArchive.h"
```
Comment (jagged-teeth) [inline Source/MissionSystem/Private/MSMissionSystem.cpp]: unused include ?

File: Source/MissionSystem/Private/ViewModels/MSObjectiveViewModel.cpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#include "ViewModels/MSObjectiveViewModel.h"
+
+#include "MSMissionObjective.h"
+#include "Templates/SubclassOf.h"
```
Comment (Frickerson) [inline Source/MissionSystem/Private/ViewModels/MSObjectiveViewModel.cpp]: <>

**Review by Frickerson — CHANGES_REQUESTED** [review]

File: Source/MissionSystem/Private/ViewModels/MSViewModel.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    auto * mission_vm = NewObject< UMSMissionViewModel >( this );
+    mission_vm->Initialize( mission );
+    ActiveMissions.Add( mission_vm );
+
+    UE_MVVM_BROADCAST_FIELD_VALUE_CHANGED( ActiveMissions );
+}
+
+void UMSViewModel::SetMissionEnded( UMSMission * mission )
```
Comment (jagged-teeth) [inline Source/MissionSystem/Private/ViewModels/MSViewModel.cpp:19]: `const UMSMission * mission` on most of these functions would maybe be beneficial

File: Source/MissionSystem/Private/MSMissionSystem.cpp (hunk 1 above)
Comment (TheEmidee) [reply Source/MissionSystem/Private/MSMissionSystem.cpp]: done

File: Source/MissionSystem/Private/ViewModels/MSObjectiveViewModel.cpp (hunk 2 above)
Comment (TheEmidee) [reply Source/MissionSystem/Private/ViewModels/MSObjectiveViewModel.cpp]: done

File: Source/MissionSystem/Private/ViewModels/MSViewModel.cpp (hunk 3 above)
Comment (TheEmidee) [reply Source/MissionSystem/Private/ViewModels/MSViewModel.cpp:19]: wont fix as this can escalate quickly with const pointers in UE

**Review by Frickerson — DISMISSED** [review]

**Review by jagged-teeth — APPROVED** [review]

**Pull request merged**